src/Symfony/Component/OptionsResolver/OptionConfigurator.php

```diff
@@ -0,0 +1,130 @@
+<?php
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver;
+
+use Symfony\Component\OptionsResolver\Exception\AccessException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+
+final class OptionConfigurator
+{
+    /**
+     * The name of the defined option
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var OptionsResolver
+     */
+    private $resolver;
+
+    public function __construct(string $name, OptionsResolver $resolver)
+    {
+        $this->name = $name;
+        $this->resolver = $resolver;
+        $this->resolver->setDefined($name);
+    }
+
+    /**
+     * Adds allowed types for this option.
+     *
+     * @param string|string[] $types One or more accepted types
+     *
+     * @return $this
```

Comment:
This annotation is not really necessary if the class is final and you already have `: self` as return typehint (same for other methods)

Reply:
`@return $this` documents that this is a fluent API (returning the same instance), so it gives more details that `self`.

Reply:
Just curious, what would those details be vs `self`?

Reply:
Note that this class is final and doesn't extend of any class either. In this case `@return $this` doesn't provide any value vs `self`, AFAIK.

Reply:
`self` says it is the same class. `$this` says it is the same *instance of the class*.

Reply:
True, but `self` also means "an object of the class" when you use it as a return type. That's why I think `@return $this` is useless here.

Reply:
`an object of the class` is not the same than `the same instance of the class`

Reply:
On this context where *there is no parent class* and *it's also `final class`* (all my comments above are based on these conditions) the result for IDE/static analyzers will be the same.

Reply:
@yceruto I think what @stof means is that returning `self` allow to return a new instance of the class in the method for example : 
```

Reply:
@fancyweb I understand what you mean, and what Stof says, but imho `@return $this` is not needed here because `: self` doesn't mean you'll get a **new instance** of the same class as the one the method is defined on, but simply an instance of the same class.

If `@return $this` helps to clarify that it is not a new instance I guess it's okay to add it.